data-node/src/main/java/org/graylog/datanode/configuration/RootCertificateFinder.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.datanode.configuration;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+
+public class RootCertificateFinder {
+
+    public X509Certificate findRootCert(Path keystorePath,
```

Comment:
As this class has no dependencies, implements no interface and uses no properties, I'd suggest to convert it into a static call. It will remove the unnecessary dependency injection and dragging of the instance all over the usage places.

Reply:
I'm strongly against it.
Static calls make it impossible to mock this class in unit tests of client classes.

Reply:
Then let's go ahead with an RootCertificateFinder interface if we want to have the flexibility and need for dependency injection.